src/mainwindow.cpp

```diff
@@ -413,6 +476,48 @@ void MainWindow::on_openFileButton_clicked()
     openFile(fileName);
 }
 
+void MainWindow::onRecordDataButtonClicked()
+{
+    ui->mainPageStack->setCurrentWidget(ui->recordDataPage);
+}
+
+void MainWindow::onHomeButtonClicked()
+{
+    ui->mainPageStack->setCurrentWidget(ui->startPage);
+}
+
+void MainWindow::onStartRecordingButtonClicked(bool checked)
+{
+    if (checked) {
+        ui->startRecordingButton->setIcon(QIcon::fromTheme(QStringLiteral("media-playback-stop")));
+        ui->startRecordingButton->setText(tr("Stop Recording"));
+
+        QStringList perfOptions;
+
+        if (!ui->callGraphComboBox->currentData().toString().isEmpty()) {
```

Comment:
again, temporary to prevent repetition in code:

```
const auto callGraphOption = ui->callGraphComboBox->currentData().toString();
if (!callGraphOption.isEmpty()) {
    perfOptions << QStringLiteral("--call-graph") << callGraphOption;
}
```